Aws/S3/Commands/CopyObject.hs

```diff
@@ -55,12 +56,12 @@ instance SignQuery CopyObject where
     signQuery CopyObject {..} = s3SignQuery S3Query {
                                  s3QMethod = Put
                                , s3QBucket = Just $ T.encodeUtf8 coBucket
-                               , s3QObject = Just $ T.encodeUtf8 coObjectName
+                               , s3QObject = Just . HTTP.urlEncode False $ T.encodeUtf8 coObjectName
                                , s3QSubresources = []
                                , s3QQuery = []
                                , s3QContentType = coContentType
                                , s3QContentMd5 = Nothing
-                               , s3QAmzHeaders = map (second T.encodeUtf8) $ catMaybes [
+                               , s3QAmzHeaders = map (second (HTTP.urlEncode False . T.encodeUtf8)) $ catMaybes [
```

Comment:
Did you verify that this is necessary? URL Encoding HTTP headers seems weird to me.